management/server/posture/os_version.go

```diff
@@ -0,0 +1,96 @@
+package posture
+
+import (
+	"github.com/hashicorp/go-version"
+	nbpeer "github.com/netbirdio/netbird/management/server/peer"
+	log "github.com/sirupsen/logrus"
+)
+
+type MinVersionCheck struct {
+	MinVersion string
+}
+
+type MinKernelVersionCheck struct {
+	MinKernelVersion string
+}
+
+type OSVersionCheck struct {
+	Android *MinVersionCheck
+	Darwin  *MinVersionCheck
+	Ios     *MinVersionCheck
+	Linux   *MinKernelVersionCheck
+	Windows *MinKernelVersionCheck
+}
+
+var _ Check = (*OSVersionCheck)(nil)
+
+func (c *OSVersionCheck) Check(peer nbpeer.Peer) (bool, error) {
+	peerGoOS := peer.Meta.GoOS
+	switch peerGoOS {
+	case "android":
+		return checkMinVersion(peerGoOS, peer.Meta.Core, c.Android)
```

Comment:
Move from `peer.Meta.Core` to `peer.Meta.OsVersion` after https://github.com/netbirdio/netbird/pull/1495